### PR TITLE
Linear regression contracts

### PR DESCRIPTION
**Pull request by wamartin-aml**
Fairly WIP PR for proving out tighter contracts in order to work more closely with other teams

### CODE REVIEW

File: src/python/interpret/api/templates.py [hunk 1]
```diff
@@ -39,10 +39,10 @@ def data(self, key=None):
         return self._internal_obj["specific"][key]
 
     def visualize(self, key=None):
-        from ..visual.plot import plot_line, plot_bar, plot_horizontal_bar
-        from ..visual.plot import sort_take, plot_pairwise_heatmap
-
+        from ..visual.plot import plot_line, plot_bar, plot_horizontal_bar, plot_horizontal_bar2
+        from ..visual.plot import get_sort_indexes, sort_take, sort_take2, plot_pairwise_heatmap
         data_dict = self.data(key)
+        mli_data = self.data(-1)["mli"]
```
Comment (hchandola) [inline src/python/interpret/api/templates.py]: keep string literals in a constants class ?

File: src/python/interpret/visual/plot.py [hunk 2]
```diff
@@ -464,6 +503,26 @@ def sort_take(
...
 
+def get_sort_indexes(data, sort_fn=None, top_n=None):
+    if top_n is None:
+        top_n = len(data)
+
+    if sort_fn is not None:
+        scored_vals = list(map(sort_fn, data))
+        return np.argsort(scored_vals)[:top_n]
+    else:
+        return np.array(range(top_n))
```
Comment (hchandola) [inline src/python/interpret/visual/plot.py]: `np.arange(top_n)`

File: src/python/interpret/api/templates.py (hunk 1 above)
Comment (hchandola) [reply src/python/interpret/api/templates.py]: is `-1` a special key for things like mli?

Comment (wamartin-aml) [reply src/python/interpret/api/templates.py]: So far this is just following current practices. If we want the codebase constantified, that should be a different PR

**Comment by interpret-ml** [comment]
Progress looks good. Thanks for working on this!

There's a major change coming to the directory structure across the package. We're going to update `data-interop` branch with the associated changes in the following few days.

**Comment by wamartin-aml** [comment]
Give our conversation yesterday, these additions for lime and shap don't include any changes to old visualization code, as we'll work toward sharing one visualization instead

**Comment by interpret-ml** [comment]
Merging into `data-interop`, will fix up some small points and then push to `develop `ready for release today. Thanks for the PR!

**Pull request merged**